gt/main.c

```diff
@@ -90,30 +186,72 @@ gt_proc(void *arg)
 			 *
 			 * Other packets will be fowarded directly.
 			 */
+			rte_pktmbuf_adj(m, sizeof(struct ether_hdr));
+
+			ip_hdr = get_ip_hdr(m, &inner_ip_ver);
+			if (ip_hdr == NULL) {
+				rte_pktmbuf_free(m);
```

Comment:
We need a warning here because there's a kind of packet that is coming from Gatekeeper servers that we don't know what to do with them. Notice that this is a very special warning due to the fact that it could potentially sign that there's a security hole in Gatekeeper, so I recommend using the log level ALERT. The message must include `inner_ip_ver`.